lib/ansible/modules/cloud/amazon/iam_user_info.py

```diff
@@ -0,0 +1,189 @@
+        create_date:
+            description: the datetime user was created
+            returned: if user exists
+            type: str
+            sample: "2016-05-24T12:24:59+00:00"
+        password_last_used:
+            description: the last datetime the password was used by user
+            returned: if password was used at least once
+            type: str
+            sample: "2016-05-25T13:39:11+00:00"
+        path:
+            description: the path to user
+            returned: if user exists
+            type: str
+            sample: "/dev/"
+        user_id:
+            description: the unique user id
+            returned: if user exists
+            type: str
+            sample: "AIDUIOOCQKTUGI6QJLGH2"
+        user_name:
+            description: the user name
+            returned: if user exists
+            type: str
+            sample: "test_user"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, ParamValidationError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def list_iam_users(connection, module):
+
+    name = module.params.get('name')
+    group = module.params.get('group')
+    path = module.params.get('path')
+
+    params = dict(MaxItems=1000)
+    iam_users = []
+
+    if name and not path:
+        params['UserName'] = name
+        try:
+            iam_users.append(connection.get_user(**params)['User'])
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    if group:
+        params['GroupName'] = group
+        try:
+            iam_users = connection.get_group(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg="Failed while listing IAM groups %s " % e.message,
+                             **camel_dict_to_snake_dict(e.response))
+        if name:
+            iam_users = [user for user in iam_users if user['UserName'] == name]
+
+    if path and not group:
+        params['PathPrefix'] = path
+        try:
+            iam_users = connection.list_users(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg="Failed while listing IAM users %s " % e.message,
+                             **camel_dict_to_snake_dict(e.response))
+        if name:
+            iam_users = [user for user in iam_users if user['UserName'] == name]
+
+    module.exit_json(iam_users=[camel_dict_to_snake_dict(user) for user in iam_users])
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(),
+            group=dict(),
+            path=dict(default='/')
+        )
+    )
+
+    module = AnsibleModule(
```

Comment:
This can be migrated to AnsibleAWSModule, which will handle things like the boto requirements and has a nice fail_json_aws method

Reply:
https://github.com/ansible/ansible/pull/23382/commits/e8d91f4a221d6d196b5d48988f55bcc61764f46f and https://github.com/ansible/ansible/pull/23382/commits/df9d52172242e86b88497c7468e9a3d25c8bf6f4 have the related updates here.